sklearn/utils/validation.py

```diff
@@ -429,6 +432,10 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None,
         msg = ("Data with input dtype %s was converted to %s%s."
                % (dtype_orig, array.dtype, context))
         warnings.warn(msg, DataConversionWarning_)
+
+    if copy and array is array_orig:
```

Comment:
I'm not sure but if array is not array_orig but array.base is array_orig and copy = true, shouldn't we create a real copy ?